modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/ViewsIntegrationTest.java

```diff
@@ -0,0 +1,432 @@
+        assertEquals(1, cache4.query(new SqlFieldsQuery("select * from my_view2")).getAll().size());
+    }
+
+    /**
+     * Tests views drop on schema drop.
+     */
+    @Test
+    public void testSchemaCleaning() throws Exception {
+        initGrids(3);
+        initTable(5);
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE1")
+            .setIndexedTypes(Integer.class, Integer.class));
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE2")
+            .setIndexedTypes(Integer.class, Integer.class)
+            .setSqlSchema("PUBLIC")
+        );
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE3")
+            .setIndexedTypes(Integer.class, Integer.class)
+            .setSqlSchema("MY_SCHEMA")
+        );
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE4")
+            .setIndexedTypes(Integer.class, String.class)
+            .setSqlSchema("MY_SCHEMA")
+        );
+
+        sql("create or replace view my_view as select * from my_table");
+        sql("create or replace view cache1.my_view as select * from public.my_table");
+        sql("create or replace view my_schema.my_view as select * from public.my_table");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 1);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 1);
+
+        client.destroyCache("CACHE1");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 0);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 1);
+
+        client.destroyCache("CACHE2");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 0);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 1);
+
+        client.destroyCache("CACHE3");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 0);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 1);
+
+        client.destroyCache("CACHE4");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 0);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 0);
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE1")
+            .setIndexedTypes(Integer.class, Integer.class));
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE3")
+            .setIndexedTypes(Integer.class, Integer.class)
+            .setSqlSchema("MY_SCHEMA")
+        );
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 0);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 0);
+    }
+
+    /**
+     * Tests views on not existing schema.
+     */
+    @Test
+    public void testNotExistingSchema() throws Exception {
+        initGrids(3);
+        initTable(3);
+
+        assertThrows("CREATE VIEW my_schema.my_view AS SELECT * FROM public.my_table", IgniteSQLException.class,
```

Comment:
Let's check full message or at least a part from it.

Reply:
There are different messages for calcite and H2 engines, so only common part is checked here